jsettlers.logic/src/main/java/jsettlers/logic/movable/BehaviorTreeHelper.java

```diff
@@ -0,0 +1,239 @@
+    }
+
+    public static Repeat<Context> Repeat(Repeat.Policy policy, Node<Context> condition, Node<Context> child) {
+        return new Repeat<>(policy, condition, child);
+    }
+
+    public static Repeat<Context> Repeat(Node<Context> condition, Node<Context> child) {
+        return new Repeat<>(condition, child);
+    }
+
+    @SafeVarargs
+    public static Selector<Context> Selector(Node<Context>... children) {
+        return new Selector<>(children);
+    }
+
+    @SafeVarargs
+    public static Sequence<Context> Sequence(Node<Context>... children) {
+        return new Sequence<>(children);
+    }
+
+    public static Succeeder<Context> Succeeder() {
+        return new Succeeder<>();
+    }
+
+    public static Wait<Context> Wait(Node<Context> condition) {
+        return new Wait<>(condition);
+    }
+
+    public static NotificationCondition NotificationCondition(Class<? extends Notification> type) {
+        return new NotificationCondition(type);
+    }
+
+    public static NotificationCondition NotificationCondition(Class<? extends Notification> type, boolean consume) {
+        return new NotificationCondition(type, consume);
+    }
+
+    public static Node<Context> WaitForTargetReached_FailIfNotReachable() {
+        // wait for targetReached, but abort if target not reachable
+        return Sequence(
+            Inverter(NotificationCondition(SteeringComponent.TargetNotReachedTrigger.class, true)),
+            Wait(NotificationCondition(SteeringComponent.TargetReachedTrigger.class, true))
+        );
+    }
+
+    public static Property<Context, Boolean> SetAttackableWhile(boolean value, Node<Context> child) {
+        return new Property<>(
+            (c,v)->{c.entity.attC().IsAttackable(v);},
+            (c)->{return c.entity.attC().IsAttackable();},
+            value,
+            child);
+    }
+
+    public static Property<Context, Boolean> SetIdleBehaviorActiveWhile(boolean value, Node<Context> child) {
+        return new Property<>(
+            (c,v)->{c.entity.steerC().IsIdleBehaviorActive(v);},
+            (c)->c.entity.steerC().IsIdleBehaviorActive(),
+            value,
+            child);
+    }
+
+    public static Guard<Context> TriggerGuard(Class<? extends Notification> type, Node<Context> child) {
+        return new Guard<>(entity -> entity.comp.getNotificationsIt(type).hasNext(), true, child);
+    }
+
+    public static Action<Context> StartAnimation(EMovableAction animation, short duration) {
+        return new Action<>(c -> {
+            c.entity.aniC().startAnimation(animation, duration);
+        });
+    }
+
+    public static void convertTo(Entity entity, EMovableType type) {
+        Entity blueprint = EntityFactory.CreateEntity(entity.gameC().getMovableGrid(), type, entity.movC().getPos(), entity.movC().getPlayer());
+        entity.convertTo(blueprint);
+    }
+
+    public static <T> Node<T> Optional(Node<T> child) {
```

Comment:
This should have a different name. An Optional is a fixed thing in java8, which makes this confusing.